features/support/env.js

```diff
@@ -1,20 +1,33 @@
-const { setWorldConstructor, After, setDefaultTimeout } = require('cucumber');
+const { setWorldConstructor, After, setDefaultTimeout, Status } = require('cucumber');
 const { Builder } = require('selenium-webdriver');
 const firefox = require('selenium-webdriver/firefox');
 
 class CustomWorld {
-  constructor() {
+  constructor({ attach }) {
+    this.attach = attach;
     this.driver = new Builder()
       .forBrowser('firefox')
-      .setFirefoxOptions(
-        new firefox.Options().headless()
-      ).build();
+      .setFirefoxOptions(this.firefoxOption())
+      .build();
+  }
+
+  firefoxOption() {
+    // GitHub Actions set CI to true
+    // Ref: https://docs.github.com/en/free-pro-team@latest/actions/reference/environment-variables#default-environment-variables
+    return process.env.CI ? new firefox.Options().headless() : new firefox.Options()
   }
 }
 
 setWorldConstructor(CustomWorld);
 
-After(function() {
+After(function(testCase) {
+  if (testCase.result.status == Status.FAILED) {
+    return this.driver.takeScreenshot().then( screenShot => {
+      console.log('screenshot', screenShot)
+      this.attach(screenShot, 'image/png');
+      this.driver.quit()
```

Comment:
I assume this is making the driver take a screenshot when the test fails, so we have helpful test results when debugging test failures. Where is the screenshot being saved to? To a local directory ignored by git?

Reply:
Still exploring, but the intention is to generate report using `cucumber-html-reporter`.
And it will be a separate PR.

Reply:
I was trying to avoid making test report but somehow the test is failing on CI only and I need a way to view the screenshot to debug.